creator-node/src/services/stateMachineManager/stateReconciliation/issueSyncRequest.jobProcessor.js

```diff
@@ -259,7 +281,9 @@ const _additionalSyncIsRequired = async (
       logger.info(`${logMsgString} secondaryClock ${secondaryClockValue}`)
 
       // Record starting and current clock values for secondary to determine future action
-      if (initialSecondaryClock === null) {
+      if (syncMode === SYNC_MODES.MergePrimaryAndSecondary) {
+        initialSecondaryClock = 0
```

Comment:
1. why do we do this?
2. wouldn't it be reset to -1 earlier in the MergePrimaryAndSecondary process?
3. what if the initial clearing of the secondary failed and its clock value isn't actually 0 so it looks like it progresses on the next check? shouldn't we use `secondaryClockValue` from above as the initial value to make sure it actually progresses from that?

Reply:
MErgePrimaryAndSecondary does 2 things
- it first syncs primary from secondary, then it just calls the regular syncSecondaryFromPrimary code (including this function) with a forceResync flag
- meaning before this function call, secondary clock was reset - it's reset to 0 btw, not 1 - and synced frmo thre

your 3rd point is a good one and one that i considered - issue is this function's purpose is to know if secondary sync failed or not, so there is no prior or other way to know. it's a shitty system currently. my thinking here is that i'd rather compare if it progressed from 0 bc the scenarios are:
1. secondary was correctly reset to 0, and then had a successful sync -> this logic would correctly enqueue another sync if needed, or do nothing if caught up
2. secondary was correctly reset to 0, but failed to sync from there -> this logic would correctly enqueue another sync
3. secondary was not correctly reset to 0 -> this logic wouldn't really apply since we are still in a divergent state, it would either issue another sync or not, either way its bad, but subsequent divergent state check will rectify eventually

def this state is not ideal - i just want to roll this out as a very cautious first step and then improve logic from there. felt like redoing this would take some time

but i'd love your input here - what do you think we should do?